engine/src/main/java/org/camunda/bpm/engine/impl/cmmn/entity/repository/CaseDefinitionQueryImpl.java

```diff
@@ -227,6 +234,19 @@ public void checkQueryOk() {
     }
   }
 
+  private boolean checkCmmnEnabled() {
+      if (cmmnEnabled == null) {
+          ProcessEngineConfigurationImpl configuration = Context.getProcessEngineConfiguration();
+          if (configuration != null) {
+              cmmnEnabled = Boolean.valueOf(configuration.isCmmnEnabled());
+          } else {
+              cmmnEnabled = Boolean.TRUE;
+          }
+          
+      }
+      return cmmnEnabled.booleanValue();
```

Comment:
Why do we need the `cmmnEnabled` flag? A new instance of the `CaseDefinitionQueryImpl` class is created when the appropriate `RepositoryService` method is called. Wouldn't the following be enough:

```suggestion
      return Context.getCommandContext()
          .getProcessEngineConfiguration()
          .isCmmnEnabled();
```